mm/ktsan/ktsan.h

```diff
@@ -89,6 +112,27 @@ struct kt_stack_s {
 	int			size;
 };
 
+/* Stack depot. */
+
+struct kt_stack_depot_obj_s {
+	kt_stack_depot_obj_t	*link;
+	kt_stack_t		stack;
+};
+
+struct kt_stack_depot_part_s {
+	kt_stack_depot_obj_t	*head;
+	u32			hash;
```

Comment:
how do you use it?
you need to push hash into kt_stack_depot_obj_s and compare it before comparing the pc array

then, kt_stack_depot_part_s will become a single field, provided that you embed parts into kt_stack_depot_s, you can get rid of kt_stack_depot_part_s altogether and do:

struct kt_stack_depot_s {
   ...
   kt_stack_depot_obj_t    *parts[...];